### PR TITLE
Implemented payload size limit

### PR DESCRIPTION
**Pull request by supreme2580**
Closes: https://github.com/zig-bitcoin/btczee/issues/114

### CODE REVIEW

File: src/network/wire/lib.zig [hunk 1]
```diff
@@ -102,6 +102,14 @@ pub fn receiveMessage(
     const payload_len = try r.readInt(u32, .little);
     const checksum = try r.readBytesNoEof(4);
 
+    // Calculate total message size
+    const precomputed_total_size = 24; // network (4 bytes) + command (12 bytes) + payload size (4 bytes) + checksum (4 bytes)
+    const total_message_size = precomputed_total_size + payload_len;
+
+    if (total_message_size > MAX_SIZE) {
+        return error.InvaliPayloadLen;
+    }
```
Comment (tdelabro) [inline src/network/wire/lib.zig:111]: Can you factorize this inside a function? For both read and write. That takes `payload_len` as argument and return `!void`

Comment (tdelabro) [reply src/network/wire/lib.zig:111]: And `const MAX_SIZE: usize = 0x02000000; // 32 MB` can be inside this function scope

**Review by tdelabro — CHANGES_REQUESTED** [review]

File: src/network/wire/lib.zig (hunk 1 above)
Comment (supreme2580) [reply src/network/wire/lib.zig:111]: Okay boss will do this now

Comment (supreme2580) [reply src/network/wire/lib.zig]: @tdelabro I made send and receive message return !void and if the payload size is too large it returns InvalidPayloadLen

Comment (supreme2580) [reply src/network/wire/lib.zig]: @tdelabro also just fixed the typo from InvaliPayloadLen to InvalidPayloadLen

**Review by tdelabro — APPROVED** [review]

**Pull request merged**